src/utils.ts

```diff
@@ -492,14 +492,10 @@ function patchMatrixClientForRetry() {
         return cb(...result);
       } catch (err) {
         // Need to retry.
-        let retryAfterMs = attempt * attempt * REQUEST_RETRY_BASE_DURATION_MS;
-        if ("retry_after_ms" in err) {
-          try {
-            retryAfterMs = Number.parseInt(err.retry_after_ms, 10);
-          } catch (ex) {
-            // Use default value.
-          }
-        }
+        const retryAfterMs =
+          "retryAfterMs" in err
+            ? err.retryAfterMs
+            : attempt * attempt * REQUEST_RETRY_BASE_DURATION_MS;
         LogService.debug(
           "Draupnir.client",
           `Waiting ${retryAfterMs}ms before retrying ${params.method} ${params.uri}`
```

Comment:
I wonder if we should log something if we can't extract `retryAfterMs` from the bot-sdk version of the wrapped error at all?

Reply:
It's my understanding that if it's not present in the bot SDK, either the homeserver didn't return any ratelimiting info (aside from the 429 itself) at all, or it responded with a malformed `retry-after` header *and* didn't include the deprecated `.retry_after_ms` field in JSON. I don't think we really need to account for such an extreme edge case since I doubt it's remotely likely to happen

Reply:
The issue is that we don't know if all of our code correctly converts errors to matrix-bot-sdk's `MatrixError`, because of the hacks that we have to do in this file (the utils file) and in matrix-protection-suite-for-matrix-bot-sdk. So I do worry that our own code can complicate things. Either way i think i have covered my concern here https://github.com/the-draupnir-project/Draupnir/pull/694#discussion_r1921607131 :smile_cat:

Reply:
okay yeah that makes sense, I'll use your suggested changes ^.^